synapseclient/__main__.py

```diff
@@ -550,20 +561,30 @@ def build_parser():
             help='Synapse ID of project or folder where file/folder will be moved ')
     parser_mv.set_defaults(func=move)
 
-    # parser_cp = subparsers.add_parser('cp',
-    #         help='Copies a file in Synapse')
-    # parser_cp.add_argument('--id', metavar='syn123', type=str, required=True,
-    #         help='Id of entity in Synapse to be copied.')
-    # parser_cp.add_argument('--parentid', '--parentId', '-parentid', '-parentId', metavar='syn123', type=str, required=True, dest='parentid',
-    #         help='Synapse ID of project or folder where file will be moved ')
-    # parser_cp.add_argument('-v', '--version', metavar='VERSION', type=int, default=None,
-    #         help='Synapse version number of entity to retrieve. Defaults to most recent version.')
-    # parser_cp.add_argument('--setProvenance', metavar='MODE', type=str, default='traceback',
-    #         help=('Has three values to set the provenance of the copied entity-'
-    #                     'traceback: Sets to the source entity'
-    #                     'existing: Sets to source entity\'s original provenance (if it exists)'
-    #                     'None/none: No provenance is set'))
-    # parser_cp.set_defaults(func=copy)
+    parser_cp = subparsers.add_parser('cp',
+            help='Copies a file in Synapse')
+    parser_cp.add_argument('id', metavar='syn123', type=str,
+            help='Id of entity in Synapse to be copied.')
+    parser_cp.add_argument('--destinationId', metavar='syn123', type=str,
+            help='Synapse ID of project or folder where file will be copied to.  If no destinationId specified, a new project is created')
+    parser_cp.add_argument('--version','-v', metavar='1', type=int, default=None,
+            help='Synapse version number of entity to retrieve. Defaults to most recent version.')
```

Comment:
Version is a tricky one.  I think it only truly belongs to Files.  copyFile is the only function that uses version.  Should an error be thrown for all the other Entities or just Project/Folder?  The only worry would be for Project and Folders where there may be a possibility of people copying files within the projects that don't have the specified version.